### ISSUE TITLE
please add related resources to github

### ISSUE THREAD

**Issue by dsavchenko** [oda-hub/frontend-chart#8]
at least git@gitlab.astro.unige.ch:cdci/frontend/drupal7-db-for-astrooda.git is referred in make.sh but not accessible from outside

It will also be good to have container spec for the frontend to experiment with custom builds (the one on dockerhub is half-year-old)

**Comment by volodymyrss** [comment]
the frontend has just been delivered by @motame , in principle both container and chart exist and are used, but some issues might arise. I will open them as they appear to make some marginal sense.

**Comment by dsavchenko** [comment]
Nice. I will try to use it

**Comment by volodymyrss** [comment]
eh, looks like some docs written on github only were lost. do you have a local copy?.. I do not know if I do, I need to search

**Comment by dsavchenko** [comment]
> eh, looks like some docs written on github only were lost. do you have a local copy?.. I do not know if I do, I need to search

What docs are you talking about? I have nothing related to the frontend locally.

**Comment by dsavchenko** [comment]
oh, now I see. Unfortunately, not

**Comment by volodymyrss** [comment]
there was a short readme, seems like it was overwritten 4 days ago. interestingly, gitlab sync did it in a way that makes, it seems, recovery from github's reflog impossible.
 not that critical but annoying

when did you clone last? or you deleted?

**Comment by volodymyrss** [comment]
it was there from february till 4 days ago. nobody has a clone somewhere?..

**Comment by dsavchenko** [comment]
> when did you clone last? or you deleted?

I just not yet started to deploy frontend, so I was not cloning

**Comment by volodymyrss** [comment]
ah ok

**Comment by volodymyrss** [comment]
ok I found a way to restore them from github.

by the way, they might be exactly relevant for your question here:

https://github.com/oda-hub/frontend-chart

maybe you could build up the chart yourself. It only has one container. but lot's of repositories merged into it in special places.
more or less if all is put in good place all should work

**Comment by volodymyrss** [comment]
https://github.com/oda-hub/frontend-chart should not contain all that is needed

**Comment by volodymyrss** [comment]
please let me know @dsavchenko  if it works.

**Comment by dsavchenko** [comment]
> please let me know @dsavchenko if it works.

I have not tried to build container image, just used [the last one](https://hub.docker.com/layers/odahub/frontend/f914117b/images/sha256-0290b96f5aa011278801e57426738f2745d2cd91ba71a5b9c14fbba4f72e8bc3?context=explore)

What was not immediately obvious:
- The resulting page served under /cdci/astrooda/. I noticed this only after reading container logs
- Manual changes to drupal7_sites_default_settings.php.template are needed: at least _base_url_ and _reverse_proxy_addresses_ 
These may alternatively be configurable with variables and inserted by jinja


The frontend only partly works for me. At least queries are forwarded to dispatcher. But when processing is ready, no window with results appear. 

Other minor issues: 
- unnecessary query types in anatares; 
- the value of radius parameter is not validated in frontend for antares (should it?); 
- failed to reach local name resolver server. What needs to be added for it to work?

Link: https://ui.oda.virgoua.org/cdci/astrooda/ (spi_acs and antares spectrum should work)

**Comment by volodymyrss** [comment]
> > please let me know @dsavchenko if it works.
> 
> I have not tried to build container image, just used [the last one](https://hub.docker.com/layers/odahub/frontend/f914117b/images/sha256-0290b96f5aa011278801e57426738f2745d2cd91ba71a5b9c14fbba4f72e8bc3?context=explore)
> 
> What was not immediately obvious:
> 
>     * The resulting page served under /cdci/astrooda/. I noticed this only after reading container logs

Good that you figured it out! I added a note [there](https://github.com/oda-hub/frontend-chart/blob/master/README.md).

> 
>     * Manual changes to drupal7_sites_default_settings.php.template are needed: at least _base_url_ and _reverse_proxy_addresses_
>       These may alternatively be configurable with variables and inserted by jinja

Drupal should be able to work without this base_url at least, it should derive the right name from http headers. I did not not look closer.
If one location will be stored, it will not work if there are several ingresses, which is unfortunate.


> 
> 
> The frontend only partly works for me. At least queries are forwarded to dispatcher. But when processing is ready, no window with results appear.

seems like this issue https://github.com/oda-hub/frontend-astrooda/issues/5
you can notice the cause by looking into browser dev console.

I will check that the last version works.

And/or you man need to do `drush cc` - see there in make.sh

> 
> Other minor issues:
> 
>     * unnecessary query types in anatares;
> 
>     * the value of radius parameter is not validated in frontend for antares (should it?);

Makes sense, please open issues!

> 
>     * failed to reach local name resolver server. What needs to be added for it to work?

yeah, we need to connect it https://github.com/oda-hub/oda-chart/issues/12.
Or release it too, was not done yet. But it can be referenced, I will do it.

> 
> 
> Link: https://ui.oda.virgoua.org/cdci/astrooda/ (spi_acs and antares spectrum should work)

Great!

**Comment by dsavchenko** [comment]
> And/or you man need to do `drush cc` - see there in make.sh

I did it right after installation

**Comment by dsavchenko** [comment]
> seems like this issue [oda-hub/frontend-astrooda#5](https://github.com/oda-hub/frontend-astrooda/issues/5)
> you can notice the cause by looking into browser dev console.

`Uncaught TypeError: data.image.image is undefined` -- this was the error in my case. Which is another issue. 
Dispatcher response indeed has a slightly different structure, not dict-in-dict, but dict-in-list-in-dict.

But it was not the recent dispatcher I used.  After update I reproduce [oda-hub/frontend-astrooda#5](https://github.com/oda-hub/frontend-astrooda/issues/5)

**Comment by volodymyrss** [comment]
> > seems like this issue [oda-hub/frontend-astrooda#5](https://github.com/oda-hub/frontend-astrooda/issues/5)
> > you can notice the cause by looking into browser dev console.
> 
> `Uncaught TypeError: data.image.image is undefined` -- this was the error in my case. Which is another issue.
> Dispatcher response indeed has a slightly different structure, not dict-in-dict, but dict-in-list-in-dict.
> 
> But it was not the recent dispatcher I used. After update I reproduce [oda-hub/frontend-astrooda#5](https://github.com/oda-hub/frontend-astrooda/issues/5)

what's strange is that I saw exactly [oda-hub/frontend-astrooda#5](https://github.com/oda-hub/frontend-astrooda/issues/5) on your instance.

Is this paramtrs typo really in the container you use? Just grep /var/www/astrooda
There might be also actual git directory there, to check (should be removed if there, by  the way).

**Comment by dsavchenko** [comment]
git directories are there. looking into, it seems that there are proper commits

but paramters typo is still there in the container

**Comment by volodymyrss** [comment]
I realized your instance uses this:

`https://ui.oda.virgoua.org/cdci/astrooda/sites/all/modules/astrooda.saved/js/astrooda.instrument.js`

instead of this:

`https://ui.oda.virgoua.org/cdci/astrooda/sites/all/modules/astrooda/js/astrooda.instrument.js`

`astrooda.saved` is not really used in this case. And it's not needed there.
I do not really know why our instance picked up another version of the same module.

I removed it now.

please use tag `715e7e71`

**Comment by dsavchenko** [comment]
> please use tag `715e7e71`

It works! 
Also, there are no checkboxes here to select the resolver and the external resolver just works.

**Comment by volodymyrss** [comment]
I close but please reopen if unclear. Or even better make another issues per purpose.
